packages/theme-language-server-common/src/server/startServer.ts

```diff
@@ -88,7 +88,17 @@ export function startServer(
   const getTranslationsForURI: GetTranslationsForURI = async (uri) => {
     const rootURI = await findThemeRootURI(uri);
     const theme = documentManager.theme(rootURI);
-    return useBufferOrInjectedTranslations(getDefaultTranslationsFactory, theme, rootURI);
+
+    const getTranslationsFactory = (rootUri: string) => {
+      return async () => {
+        const defaultTranslations = await getDefaultTranslationsFactory(rootUri)();
+        const shopifyTranslations = await themeDocset.systemTranslations();
```

Comment:
Minor nit since this is sequential instead of "parallel while waiting for IO." [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) would make this run in parallel :D 

```suggestion
        const [defaultTranslations, shopifyTranslations] = await Promise.all([
          getDefaultTranslationsFactory(rootUri)(),
          themeDocset.systemTranslations(),
        ]);
```